servers/jax-rs-testextension/src/main/java/org/projectnessie/jaxrs/ext/NessieJaxRsExtension.java

```diff
@@ -86,15 +90,23 @@ public void beforeAll(ExtensionContext extensionContext) {
   }
 
   @Override
-  public void afterAll(ExtensionContext extensionContext) {
-    // NOP - EnvHolder will be closed by JUnit5
+  public void afterEach(ExtensionContext extensionContext) throws Exception {
+    EnvHolder env = extensionContext.getStore(NAMESPACE).get(EnvHolder.class, EnvHolder.class);
+    env.resetSecurityContext();
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) throws Exception {
```

Comment:
nit: do we need to do the same thing in beforeeach + aftereach?

Reply:
Call me overly cautious ;)